core/src/main/java/nu/yona/server/subscriptions/service/GoalIdMapping.java

```diff
@@ -18,12 +18,15 @@
 	private final UserDto user;
 	private final Set<UUID> userGoalIds;
 	private final Map<UUID, UUID> goalIdToBuddyIdmapping;
+	private final Map<UUID, UUID> buddyIdToUserIdmapping;
 
-	private GoalIdMapping(UserDto user, Set<UUID> userGoalIds, Map<UUID, UUID> goalIdToBuddyIdmapping)
+	private GoalIdMapping(UserDto user, Set<UUID> userGoalIds, Map<UUID, UUID> goalIdToBuddyIdmapping,
```

Comment:
Isn't it more logical to just move the methods provided on `GoalIdMapping` to `UserDto` and use the `UserDto` instead? Otherwise I am afraid the `GoalIdMapping` gets hard to grasp as a concept.